### PR TITLE
Add support for persistent request counters

### PR DESCRIPTION
**Pull request by tylerferrara**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
This change adds the option to enable persistence when creating request counters. When enabled, the number of requests & timestamp will never reset. This acts as an indefinite log of all HTTP requests the auditor has seen since startup.

#### Which issue(s) this PR fixes:
Part of #358 

#### Does this PR introduce a user-facing change?
NONE

```release-note
Adding support for persistent request counters which do not reset.
```
cc: @listx @amwat @justaugustus @kubernetes-sigs/release-engineering

### CODE REVIEW

File: legacy/reqcounter/reqcounter.go [hunk 1]
```diff
@@ -29,10 +29,11 @@ import (
...
 type RequestCounter struct {
-	Mutex    sync.Mutex    // Lock to prevent race-conditions with concurrent processes.
-	Requests uint64        // Number of HTTP requests since recording started.
-	Since    time.Time     // When the current request counter began recording requests.
-	Interval time.Duration // The duration of time between each log.
+	Mutex      sync.Mutex    // Lock to prevent race-conditions with concurrent processes.
+	Requests   uint64        // Number of HTTP requests since recording started.
+	Since      time.Time     // When the current request counter began recording requests.
+	Interval   time.Duration // The duration of time between each log.
+	Persistent bool          // Determine if the request counter should ever reset.
```
Comment (listx) [inline legacy/reqcounter/reqcounter.go:36]: Nit: I think the comment is easier to read if you just tell the reader what will happen. E.g.,

```
// If true, the Requests field is never set back to 0 when reset() is called.
```

Comment (listx) [reply legacy/reqcounter/reqcounter.go:36]: Actually, maybe `Resettable` is a better name? It's more to the point and is almost self-explanatory.

Comment (tylerferrara) [reply legacy/reqcounter/reqcounter.go]: Ack.

**Comment by listx** [comment]
Thanks!

/lgtm
/approve

**Pull request merged**